src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/ResourceExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Hl7.Fhir.Model;
+
+namespace Microsoft.Health.Fhir.Api.Features.Resources
+{
+    public static class ResourceExtensions
+    {
+        private const string ChildrenPropertyName = "Children";
+
+        public static IEnumerable<T> GetAllChildren<T>(
+            this Resource resource)
+        {
+            var stack = new Stack<(PropertyInfo propertyInfo, object baseObject)>(resource.GetType().GetProperties().Where(x => x.Name == ChildrenPropertyName).Select(x => (x, resource as object)));
```

Comment:
Seems like only 1 property should have the name `Children`?

Reply:
I read further down. I think this could also work as a recursive method with `GetAllChildren<T>(this Base resource)`. Which could then `foreach(r in b.Children)` (since this seems to be the only property of interest), I think we could keep static typing for this

Reply:
Yep, I was able to refactor this into a recursive method.